src/Database/Type/DateTimeType.php

```diff
@@ -301,37 +303,23 @@ public function marshal($value): ?DateTimeInterface
     }
 
     /**
-     * @param \DateTimeInterface $date DateTime object
-     * @param mixed $value Request data
-     * @return bool
-     */
-    protected function _compare($date, $value): bool
-    {
-        foreach ((array)$this->_format as $format) {
-            if ($date->format($format) === $value) {
-                return true;
-            }
-        }
-
-        return false;
-    }
-
-    /**
-     * Sets whether or not to parse dates passed to the marshal() function
-     * by using a locale aware parser.
+     * Sets whether or not to parse strings passed to `marshal()` using
+     * the locale-aware format set by `setLocaleFormat()`.
+     *
+     * This is not used by any other parser or conversion than `marshal()`.
      *
      * @param bool $enable Whether or not to enable
      * @return $this
      */
     public function useLocaleParser(bool $enable = true)
     {
         if ($enable === false) {
-            $this->_useLocaleParser = $enable;
+            $this->_useLocaleMarshal = $enable;
 
             return $this;
         }
-        if (method_exists($this->_className, 'parseDateTime')) {
-            $this->_useLocaleParser = $enable;
+        if ($this->_datetimeInstance instanceof \Cake\I18n\I18nDateTimeInterface) {
```

Comment:
Please add a "use" statement for the interface and use `I18nDateTimeInterface::class` here.